### PR TITLE
Make API more like Ruby => Make our tests more beautifuller! 

### PR DESCRIPTION
**Pull request by adrianna-chang-shopify**
This PR doesn't introduce any new behaviour, just cleans up the existing code and makes our tests more readable.

Closes: https://github.com/adrianna-chang-shopify/ruby-object-model/issues/15

Follow up to [Tom's comment](https://github.com/adrianna-chang-shopify/ruby-object-model/pull/20#pullrequestreview-826216578):
> There’s still a lot of messy ceremony in the tests around conditionally using the toy_ prefix on all the method names for the toy objects but not on the real ones. Remind me: why do we do that?

Originally, we chose to prefix all the methods in toyland with `toy_` because we were using _real_ Ruby classes as the basis for our toy objects. We didn't want to mix up behaviour that we needed to implement with behaviour coming from the _real_ world of Ruby, so we decided to give the toyland methods distinctive names.

However, now that we're using `BasicObject`s as the basis for all our toyland objects, we don't need to worry about mixing up behaviour, because `BasicObject` doesn't actually implement any of the behaviour we're defining 😄  Consequently, we can rename our toyland methods to match the real Ruby API, dropping the `toy_` prefix. Then, we can clean up our tests to call methods directly on an object, rather than having to use our confusing `call_method` with a prefix.

The TL;DR of this PR is that it:
- Renames methods in our toy objects to use real Ruby names
- Drops use of `call_method` from tests, calling the method directly on the object instead
- Cleans up the meta-programming loops in each test file that run the tests against different objects that play the role being tested => instead of using a generic `klass` variable, we use a variable that indicates the role being tested in the test suite (ie. `_Object` for object tests)
- Removes the helper module and `include`s since we no longer need this
- I also took the liberty to address https://github.com/adrianna-chang-shopify/ruby-object-model/issues/15 and rename the test files to match the behaviour being tested rather than being focused on testing "toy" behaviour

### CODE REVIEW

File: test/toy_object_test.rb [hunk 1]
```diff
@@ -3,81 +3,77 @@
...
+        assert_equal [:@bar, :@baz], @object.instance_variables
       end
     end
 
     describe "#instance_variable_defined?" do
       it "returns true if instance variable is set" do
-        call_method(@object, meth_prefix, :instance_variable_set, :@bar, :foo)
-        assert call_method(@object, meth_prefix, :instance_variable_defined?, :@bar)
+        @object.instance_variable_set(:@bar, :foo)
+        assert @object.instance_variable_defined?(:@bar)
```
Comment (tomstuart) [inline test/toy_object_test.rb]: I was going to chime in with “we could use `assert_predicate` here now” (cf `assert_kind_of`), but we can’t, because `assert_predicate` doesn’t support predicates which take an argument 🤷🏻‍♂️

**Review by tomstuart — APPROVED** [review]
Definitely more beautifuller! Thanks, this was a very therapeutic refactoring both to witness and to review 😅

File: test/toy_object_test.rb [hunk 2]
```diff
@@ -3,81 +3,77 @@
 require "toy_module"
 require "toy_class"
 
-[ 
-  [Object], [ToyObject, "toy_"],
-  [Module], [ToyModule, "toy_"],
-  [Class], [ToyClass, "toy_"]
-].each do |klass, meth_prefix|
-  describe klass do
+[Object, ToyObject, Module, ToyModule, Class, ToyClass].each do |_Object|
```
[the remark this answers is not in the document]
Comment (adrianna-chang-shopify) [reply test/toy_object_test.rb]: This makes a lot of sense! TBH I'm not even sure my argument about lack of explicitness holds anymore -- with the namespace prepended it should be relatively easy to figure out if something is coming from real Ruby or the `Toy::` namespace. I think I'm feeling ready to tackle this 😉

[the remark this answers is not in the document]
Comment (adrianna-chang-shopify) [reply test/toy_object_test.rb]: (I'll do so in a follow-up!)

[the remark this answers is not in the document]
Comment (tomstuart) [reply test/toy_object_test.rb]: 😱
